core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -383,21 +383,25 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
       const auto shmem_begin  = m_shmem_begin;
       const auto shmem_size   = m_shmem_size;
       const auto scratch_size = m_scratch_size[1];
-      auto team_lambda        = [=](sycl::group<1> g) {
+
+      sycl::accessor<char, 1, sycl::access::mode::read_write,
+                     sycl::access::target::local>
+          team_scratch_memory(sycl::range<1>(std::max(shmem_size, 0)), cgh);
```

Comment:
Why are you doing `std::max(shmem_size, 0)`?

Reply:
Safeguarding against `shmen_size` being negative by any chance, e.g. when no scratch memory is requested, resulting in a tremendous amount of memory being allocated.

Reply:
> e.g. when no scratch memory is requested

Then it would be set to zero
https://github.com/kokkos/kokkos/blob/5a0b850ca52a60722e5cd5efe65264be72f50ac6/core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp#L233

This does not feel like the right place for check invariants of your class.  Also that probably something you should not be doing silently.

Reply:
I simply removed it.